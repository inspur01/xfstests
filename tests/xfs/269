#! /bin/bash
# FS QA Test No. 269
#
# Check that attr_list_by_handle copies the cursor back to userspace.
#
#-----------------------------------------------------------------------
# Copyright (c) 2016, Oracle and/or its affiliates.  All Rights Reserved.
#
# This program is free software; you can redistribute it and/or
# modify it under the terms of the GNU General Public License as
# published by the Free Software Foundation.
#
# This program is distributed in the hope that it would be useful,
# but WITHOUT ANY WARRANTY; without even the implied warranty of
# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
# GNU General Public License for more details.
#
# You should have received a copy of the GNU General Public License
# along with this program; if not, write the Free Software Foundation,
# Inc.,  51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
#-----------------------------------------------------------------------

seq=`basename "$0"`
seqres="$RESULT_DIR/$seq"
echo "QA output created by $seq"

here=`pwd`
tmp=/tmp/$$
status=1    # failure is the default!
trap "_cleanup; exit \$status" 0 1 2 3 15

_cleanup()
{
	cd /
	rm -rf "$tmp".* $TEST_DIR/fsmap $TEST_DIR/testout
}

# get standard environment, filters and checks
. ./common/rc
. ./common/filter
. ./common/attr
. ./common/populate

# real QA test starts here
_supported_os Linux
_require_scratch
_require_test_program "attr-list-by-handle-cursor-test"

rm -f "$seqres.full"

echo "Format and mount"
_scratch_mkfs > "$seqres.full" 2>&1
_scratch_mount

echo "Stuff file with xattrs"
mkdir $SCRATCH_MNT/foo
__populate_create_attr $SCRATCH_MNT/foo 100

echo "Run test program"
./src/attr-list-by-handle-cursor-test $SCRATCH_MNT/foo

# success, all done
status=0
exit
