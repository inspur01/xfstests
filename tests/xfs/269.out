QA output created by 269
Format and mount
Stuff file with xattrs
Run test program
Test passes.
